jgrapht-ext/src/test/java/org/jgrapht/ext/DOTImporterTest.java

```diff
@@ -94,6 +100,66 @@ public void testDirectedNoLabels() throws ImportException {
       Assert.assertEquals(3, result.edgeSet().size());
 
    }
+   
+   public void testDirectedSameLabels() throws ImportException {
+       String input =   "digraph sample {\n"
+                      + "  a -> b;"
+                      + "  b -> c;\n"
+                      + "  a [ label=\"Test\"];\n"
+                      + "  b [ label=\"Test\"];\n"
+                      + "  c [ label=\"Test\"];\n"
+                      + "}";
+
+       DirectedMultigraph<String, DefaultEdge> expected
+             = new DirectedMultigraph<String, DefaultEdge>(DefaultEdge.class);
+       expected.addVertex("a");
+       expected.addVertex("b");
+       expected.addVertex("c");
+       expected.addEdge("a", "b");
+       expected.addEdge("b", "c");
+
+        DOTImporter<String, DefaultEdge> importer = new DOTImporter<String, DefaultEdge>(
+            new VertexProvider<String>()
+            {
+                @Override
+                public String buildVertex(
+                    String label,
+                    Map<String, String> attributes)
+                {
+                    return label;
+                }
+            },
+            new EdgeProvider<String, DefaultEdge>()
+            {
+                @Override
+                public DefaultEdge buildEdge(
+                    String from,
+                    String to,
+                    String label,
+                    Map<String, String> attributes)
+                {
+                    return new DefaultEdge();
+                }
+            },
+            new VertexUpdater<String>()
+            {
+                @Override
+                public void updateVertex(
+                    String vertex,
+                    Map<String, String> attributes)
+                {
+                }
+            });
+
+       DirectedMultigraph<String, DefaultEdge> result
+             = new DirectedMultigraph<String, DefaultEdge>(DefaultEdge.class);
+       importer.read(input, result);
+
+       Assert.assertEquals(expected.toString(), result.toString());
+
+       Assert.assertEquals(3, result.vertexSet().size());
+       Assert.assertEquals(2, result.edgeSet().size());
```

Comment:
If I recall correctly, these 3 asserts can be replaced by simply invoking equals on both graphs. From the equals description in AbstractBaseGraph:

Indicates whether some other object is "equal to" this graph. Returns true if the given object is also a graph, the two graphs are instances of the same graph class, have identical vertices and edges sets with the same weights.